From 763e738b16d2548c70a9154d8903458e80138a5c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Wed, 29 May 2024 09:47:48 +0200
Subject: [PATCH 0863/1695] dts: stm32: add support for I2C3 for
 stm32mp215f-dk.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Enable I2C3 in board device-tree file for PMIC.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8c4f467b2e33ce9c2997f930d1aa3728910bf45a dts: stm32: add support for I2C3 for stm32mp215f-dk.]
Change-Id: I29fe8189b8f1eed07dd5045832b8191baba629ec
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/383481
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/dts/stm32mp215f-dk.dts | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index 2da2940c8..00e6cafd4 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -53,6 +53,17 @@
 	status = "okay";
 };
 
+&i2c3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c3_pins_a>;
+	i2c-scl-rising-time-ns = <185>;
+	i2c-scl-falling-time-ns = <20>;
+	clock-frequency = <100000>;
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+};
+
 &iwdg1 {
 	timeout-sec = <32>;
 	status = "okay";
-- 
2.47.2

